doc/conf.py

```diff
@@ -53,6 +53,14 @@
 myst_substitutions = {
     "requires_python": requires_python,
     "requires_gmt": requires_gmt,
+    "lon0": "lon{sub}`0`",  # Used in projection table
+    "lat0": "lat{sub}`0`",
+    "lon1": "lon{sub}`1`",
+    "lat1": "lat{sub}`1`",
+    "lat2": "lat{sub}`2`",
+    "lonp": "lon{sub}`p`",
+    "latp": "lat{sub}`p`",
+	"url_doc_proj": "https://www.pygmt.org/latest/projections/",
```

Comment:
Since these substitutions are used only in the projection tables, I think it makes more sense to define them in the file front matter (ref:https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#substitutions-with-jinja2

Reply:
Hm. I tried to follow the instructions at https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#substitutions-with-jinja2. But I have problems understanding how defining the substitutions within the front matter should work.  The substitutions are not found and locally I get the warning `WARNING: Malformed YAML [myst.topmatter]` (see the commits https://github.com/GenericMappingTools/pygmt/pull/3363/commits/1337033eeb0d7ebb70ef91a5c588276c54f46175, https://github.com/GenericMappingTools/pygmt/pull/3363/commits/50589a1d47f808a1804b062fe04785c80dae773d). It works when defining the substitutions within the `conf.py` file using `myst_substitutions`.

Reply:
Solution see https://github.com/GenericMappingTools/pygmt/pull/3363#discussion_r1699313796.